sdk/core/core/internal/az_platform_internal.h

```diff
@@ -15,13 +14,6 @@ AZ_NODISCARD int64_t az_platform_clock_msec();
 
 void az_platform_sleep_msec(int32_t milliseconds);
 
-typedef struct az_platform_mtx az_platform_mtx;
```

Comment:
Wait. I think you're being too aggressive here. Unless we come up with a plan to not need a mutex, we still need this for the token credential (which needs to be fixed to use the mutex). I think we;'re going to need the mutex, I'd like to come up with a plan to not require core to to need the mutex structure itself; just its functions. I have a plan to do this but I'm thinking if I can improve it.

Reply:
@vhvb1989 , we talked with Jeffrey, and we figured out another way of being thread safe without mutexes (I'll introduce interlocked exchange function to PAL). You can go ahead with removing the mutex stuff - we won't be using it.